pdf/pdfcanvas.cpp

```diff
@@ -164,12 +164,14 @@ void PDFCanvas::setDocument(PDFDocument* doc)
     }
 }
 
-qreal PDFCanvas::pagePosition(int index) const
+qreal PDFCanvas::pagePosition(int index, qreal offset) const
 {
     if( d->pages.count() == 0 )
         return 0.f;
 
-    return d->pages.value( index ).rect.y();
+    const QRectF& rect = d->pages.value( index ).rect;
+    qreal minOffset = 0., maxOffset = 1.;
+    return rect.y() + rect.height() * qBound(minOffset, offset, maxOffset);
```

Comment:
Ah, finally I got your point, sorry being slow ! You're absolutely right. I like your suggestion, I'll amend the commit to change it for this.